### PR TITLE
feat: add kwargs to paginator send & reply methods

### PR DESCRIPTION
**Pull request by SamNuttall**
## Pull Request Type

- [x] Feature addition
- [ ] Bugfix
- [ ] Documentation update
- [ ] Code refactor
- [ ] Tests improvement
- [ ] CI/CD pipeline enhancement
- [ ] Other: [Replace with a description]

## Description
Adds keyword arguments to paginator send and reply methods. 
These are then passed to the respective methods to pass-through their functionality

Example of usage - sending an ephemeral paginator: 
```
paginator = Paginator.create_from_string(bot, your_content, page_size=1000)
await paginator.send(ctx, ephemeral=True)  # <-- Added ephermal=True (other code taken from docs)
```

## Changes

- Added kwargs to paginator send & reply methods
- Updated BaseContext to InteractionContext as BaseContext does not include a send method

## Python Compatibility

- [ ] I've ensured my code works on Python `3.10.x`
- [x] I've ensured my code works on Python `3.11.x`

## Checklist

- [X] I've run the `pre-commit` code linter over all edited files
- [ ] I've tested my changes on supported Python versions
- [ ] I've added tests for my code, if applicable
- [X] I've updated / added  documentation, where applicable

### CODE REVIEW

File: interactions/ext/paginators.py [hunk 1]
```diff
@@ -352,18 +352,19 @@ def to_dict(self) -> dict:
             "components": [c.to_dict() for c in self.create_components()],
         }
 
-    async def send(self, ctx: BaseContext) -> Message:
+    async def send(self, ctx: InteractionContext, **kwargs) -> Message:
```
Comment (LordOfPolls) [inline interactions/ext/paginators.py]:
```suggestion
    async def send(self, ctx: BaseContext, **kwargs) -> Message:
```

File: interactions/ext/paginators.py [hunk 2]
```diff
@@ -13,7 +13,7 @@
     ButtonStyle,
     spread_to_rows,
     ComponentCommand,
-    BaseContext,
+    InteractionContext,
```
Comment (LordOfPolls) [inline interactions/ext/paginators.py]:
```suggestion
    BaseContext,
```

**Review by LordOfPolls — CHANGES_REQUESTED** [review]
Reverted your typehint change. While yes, BaseContext is missing the send method, this is an issue with that object and should not be changed in this PR

Once this is changed, im happy for this to be merged

**Comment by SamNuttall** [comment]
Reverted the typehint change as requested - I'll take a further look into the context types so I know in future :)

Do note that your second suggestion is now outdated as pre-commit removed the InteractionContext line, though I have fixed this via a manual commit.

**Review by LordOfPolls — APPROVED** [review]

**Review by silasary — APPROVED** [review]

**Pull request merged**